x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/SearchableSnapshotAction.java

```diff
@@ -77,8 +80,14 @@ public static SearchableSnapshotAction parse(XContentParser parser) {
     private final boolean forceMergeIndex;
     @Nullable
     private final Integer totalShardsPerNode;
-
-    public SearchableSnapshotAction(String snapshotRepository, boolean forceMergeIndex, @Nullable Integer totalShardsPerNode) {
+    private final int numberOfReplicas;
```

Comment:
We should treat this as optional and `@Nullable` similar to `totalShardsPerNode`